boot/boot.go

```diff
@@ -327,3 +327,44 @@ func MarkBootSuccessful(dev Device) error {
 	}
 	return nil
 }
+
+var ErrSystemBootModeUnsupported = errors.New("system boot mode is unsupported")
+
+// bootSystemState is used for choosing the recovery system and mode for the
+// next boot.
+type bootSystemState interface {
```

Comment:
a bit unclear to me why we need an interface given there is only one impl (the difference should be taken care at the bootloader level)

Reply:
Yeah, I managed to over complicate things a bit.